buildpacks/maven/src/main.rs

```diff
@@ -0,0 +1,290 @@
+
+        let (mvn_executable, mvn_env) = match maven_mode {
+            Mode::UseWrapper => {
+                log_info("Maven wrapper detected, skipping installation.");
+
+                let maven_wrapper_path = context.app_dir.join("mvnw");
+
+                fs::set_permissions(&maven_wrapper_path, Permissions::from_mode(0o777))
+                    .map_err(MavenBuildpackError::CannotSetMavenWrapperExecutableBit)?;
+
+                (
+                    PathBuf::from("./mvnw"),
+                    maven_repository_layer
+                        .env
+                        .apply(Scope::Build, &Env::from_current()),
+                )
+            }
+            Mode::InstallVersion {
+                version,
+                warn_about_unused_maven_wrapper,
+                warn_about_default_version,
+            } => {
+                if warn_about_unused_maven_wrapper {
+                    log_unused_maven_wrapper_warning(&version);
+                }
+
+                if warn_about_default_version {
+                    log_default_maven_version_warning(&version);
+                }
+
+                log_info(format!("Selected Maven version: {}", &version));
+
+                let maven_layer = context
+                    .buildpack_descriptor
+                    .metadata
+                    .tarballs
+                    .get(&version)
+                    .cloned()
+                    .ok_or_else(|| {
+                        MavenBuildpackError::UnsupportedMavenVersion(version.clone()).into()
+                    })
+                    .and_then(|tarball| {
+                        context.handle_layer(layer_name!("maven"), MavenLayer { tarball })
+                    })?;
+
+                log_info(format!("Successfully installed Apache Maven {}", &version));
+
+                (
+                    PathBuf::from("mvn"),
+                    maven_layer.env.apply(
+                        Scope::Build,
+                        &maven_repository_layer
+                            .env
+                            .apply(Scope::Build, &Env::from_current()),
+                    ),
+                )
+            }
+        };
+
+        let maven_goals = context
+            .platform
+            .env()
+            .get("MAVEN_CUSTOM_GOALS")
+            .map_or_else(
+                || Ok(default_maven_goals()),
+                |maven_custom_goals_string| {
+                    shell_words::split(&maven_custom_goals_string.to_string_lossy())
+                        .map_err(MavenBuildpackError::CannotSplitMavenCustomGoals)
+                },
+            )?;
+
+        let mut maven_options = context
+            .platform
+            .env()
+            .get("MAVEN_CUSTOM_OPTS")
+            .map_or_else(
+                || Ok(default_maven_opts()),
+                |maven_custom_opts_string| {
+                    // Since this is a single environment variable, when users want to add multiple
```

Comment:
This is a single environment variable, therefore whenever users would want to have multiple
options, they would expect it to be split like a UNIX shell would. 
To mimic this behavior, we need to support proper escaping for options that contain spaces.